docs/sources/developers/plugins/add-support-for-adhoc-filters.md

```diff
@@ -0,0 +1,71 @@
++++
+title = "Ad-hoc filters for data source plugins"
++++
+
+# Ad-hoc filters for data source plugins
+
+If your data source supports filtering, such as `WHERE` for SQL, consider adding support for [ad-hoc filters](TODO).
```

Comment:
Do our built in SQL plugins support ad-hoc filters?  We were looking to add this to SAP Hana (SQL based) but some challenges arose:
* How to inject a filter into complex sql queries:  
  * an outer query along with subqueries with where clauses (where do we inject the filter?)
* How to get the tagKeys?  Seems this should be query based, but there is no way to pass a query (like you can with Query type variables) - this is a limitation we hit a lot
* getTagValues() doesn't accept a key/value pair (only the "text" value from the keys dropdown).  For SQL queries (and other filter apis) we often need an "id" and a display value.  So we can show a nice display value in the Select, but supply the associated id to the where clause.  Another limitation we hit a lot.

Reply:
Searching for getTagKeys tells me that Elasticsearch, InfluxDB, and Prometheus seems to be the only ones that use ad-hoc filters. Should we change the example perhaps?